Load the correct config.json and genesisblock.json based on the NODE_ENV

reasonthearchitect opened the pull request:
Currently, users must cp files from the test package in order to run the test suite. Additionally, before starting the server users must dropdb and createdb before running node app.js.

In this pull request I have created a file, on the root, called alias.js. This file should be called before all others. This file will check the NODE_ENV.toLowercase() var to see if it is of type 'test'. IF it is the file /test/config.json and /test/genesisBlock.json rather than the root directory files. If it is any other type (or not set) it will load up the root files. Once loaded, the files are cached as singletons.

Additionally, I have updated and call to require('<path>/[config || genesisBlock}.json in all files to ensure that they receive the right version of configuration for the environment.

Also, i have added better-npm-run to the project in order to write cleaner npm run commands. This is not required, however, it seems that the file becomes more legible.

Finally. the .eslintignore file I have added alias as a global. This simplifies its use, however, it can easily be created as a module export if that is more inline with your teams opinion. .gitignore has added package-lock.json.

Hope to here from you soon.
Rhys

MaciejBaj commented on test/api/transactions.js:
```diff
@@ -1,13 +1,13 @@
-var modulesLoader = require('../common/initModule').modulesLoader;
-var transactionTypes = require('../../helpers/transactionTypes.js');
-var genesisblock = require('../genesisBlock.json');
+var node 										= alias.require('test/node.js');
```
prefer the regular indents

reasonthearchitect replied:
I will make sure to follow that in the future.

MaciejBaj commented on test/api/transport/transport.blocks.js:
```diff
@@ -1,11 +1,10 @@
-var genesisblock = require('../../genesisBlock.json');
-var verify = require('../../../modules/blocks/verify.js');
-var bson = require('../../../helpers/bson.js');
+var node 					= alias.require('test/node.js');
```
prefer the regular indents

MaciejBaj commented on test/common/initModule.js:
```diff
@@ -1,26 +1,22 @@
-var Transaction = require('../../logic/transaction.js');
-var Account = require('../../logic/account.js');
-var Sequence = require('../../helpers/sequence.js');
+var Sequence 			= alias.require('helpers/sequence.js');
```
prefer the regular indents

MaciejBaj commented on test/common/initModule.js:
```diff
@@ -182,22 +178,22 @@ var modulesLoader = new function () {
-			{system: require('../../modules/system')},
-			{transactions: require('../../modules/transactions')},
-			{transport: require('../../modules/transport')}
+			{accounts: 					alias.require('modules/accounts')},
```
prefer the regular indents

reasonthearchitect replied:
I have reverted the files to use standard indentation and pushed it to the repo.

reasonthearchitect commented:
I have reverted the tabing on require() to spaces as per requested. Additionally, while writing unit tests for helpers/config.js last night, I noted that you can load another config file from the command line. As I was not aware if this is being done somewhere (production i suppose) I have done the command line check in the alias.js file. This will ensure that we are able to still override the configuration file. However, I did not see and option for genesisBlock.json overriding.

As a result, I have updated this branch with the correct logic to still ensure loading of a configuration file provided in the command line args.

reasonthearchitect commented:
Guys.... any thoughts on this one... are you still interested in it?

SargeKhan commented:
Hey @reasonthearchitect, I have reviewed the pull request and discussed it with @willclarktech, and I think that we shouldn't follow `alias.rewire` the approach of importing files. The rationale behind it is that this solution adds unnecessary obscurity in our code, because we ask `alias.rewire` to implicitly decide something for us, as opposed to us being more explicit in our code.

I really like the idea of using different configuration and genesis block based on ENV variable, and I would be glad to merge it but not the `alias.require` part of this PR. Let me know if you have any questions. 

Thanks for your time and great effort!

SargeKhan commented:
As discussed on gitter, I will close this PR and @reasonthearchitect will open a new PR which incorporates the changes which impact only config.json and genesisBlock.json
